src/db/entity/hub/challenge.ts

```diff
@@ -1,19 +1,38 @@
 import { Entity, Column, PrimaryGeneratedColumn } from "typeorm";
+import { Cacheable } from "../../../util/cache";
+import { IsDefined, MaxLength } from "class-validator";
 
 @Entity ()
-export class Challenge {
+export class Challenge implements Cacheable {
   @PrimaryGeneratedColumn()
   id: number;
 
+  @IsDefined({ message: "The title must be defined" })
+  @MaxLength(128)
   @Column("varchar", { length: 128 })
   title: string;
 
+  @IsDefined({ message: "The description must be defined" })
+  @MaxLength(500)
   @Column("varchar", { length: 500 })
   description: string;
 
+  @IsDefined({ message: "The company must be defined" })
+  @MaxLength(128)
   @Column("varchar", { length: 128 })
   company: string;
 
+  @MaxLength(255)
   @Column("varchar", { length: 255 })
```

Comment:
If we already have the constraint on the `@Column` annotation do we need another annotation for the same constraint?

Reply:
The @Column annotation is used by TypeORM when creating the DB table and the @MaxLength annotation is used by the validation library.
I suppose having them both is not great but we'd have to do all of the validation manually otherwise.

Reply:
Okay, that makes sense